templates/create-daml-app/ui/src/components/PartyListEdit.tsx

```diff
@@ -7,15 +7,17 @@ import { Party } from '@daml/types';
 
 type Props = {
   parties: Party[];
+  partyToAlias: Map<Party,String>;
   onAddParty: (party: Party) => Promise<boolean>;
 }
 
 /**
  * React component to edit a list of `Party`s.
  */
-const PartyListEdit: React.FC<Props> = ({parties, onAddParty}) => {
+const PartyListEdit: React.FC<Props> = ({parties, partyToAlias, onAddParty}) => {
   const [newParty, setNewParty] = React.useState('');
   const [isSubmitting, setIsSubmitting] = React.useState(false);
+  const options = Array.from(partyToAlias.entries()).map((e) => ({key: e[0], text: e[1], value: e[0]}))
```

Comment:
maybe it’s worth factoring out the function that maps from (username, partyid) to the `Form.select` options.

Reply:
I factored it out.